commonmark-ext-metadata/src/main/java/org/commonmark/ext/metadata/MetadataVisitor.java

```diff
@@ -0,0 +1,29 @@
+package org.commonmark.ext.metadata;
+
+import org.commonmark.node.AbstractVisitor;
+import org.commonmark.node.CustomNode;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MetadataVisitor extends AbstractVisitor {
+    private Map<String, List<String>> data;
+
+    public MetadataVisitor() {
+        data = new HashMap<>();
```

Comment:
How about using a `LinkedHashMap` instead? I'm not sure if YAML generally preserves the order of key-value pairs or not, but it seems like it would be appropriate here.